### PR TITLE
[plugin-rest-api] Add ability to define 'STEP' cookie store level

### PR DESCRIPTION
**Pull request by abudevich**

### CODE REVIEW

File: docs/modules/plugins/pages/plugin-rest-api.adoc [hunk 1]
```diff
@@ -36,6 +36,13 @@ a|`true`
 |`false`
 |Enable logging of HTTP request/response headers and bodies (applied to the following content types only: `text/*`, `application/json`, `application/xml`)
 
+|`rest-api.http.cookie-store-level`
+a|`global`
+`story`
+`scenario`
+|`global`
+|The property that allows to configure level of storing cookies during execution requests.
```
Comment (valfirst) [inline docs/modules/plugins/pages/plugin-rest-api.adoc:44]: please provide more information for each level value.

Comment (abudevich) [reply docs/modules/plugins/pages/plugin-rest-api.adoc]: updated, thanks

File: docs/modules/plugins/pages/plugin-rest-api.adoc [hunk 2]
```diff
@@ -36,6 +36,19 @@ a|`true`
...
 
+|`rest-api.http.cookie-store-level`
+a|`global`
+`story`
+`scenario`
+`step`
+|`global`
+a|The property that allows to configure level of storing cookies during execution requests:
+
+* `global` - cookies, set during any test, are available in all subsequent tests until the entire test suite completes or the cookies are explicitly cleared.
```
Comment (valfirst) [inline docs/modules/plugins/pages/plugin-rest-api.adoc:47]: it makes sense to mention the cookies are shared across multiple threads

File: vividus-plugin-rest-api/src/main/java/org/vividus/http/CookieStoreProvider.java [hunk 3]
```diff
@@ -39,6 +39,14 @@ public CookieStore getCookieStore()
         return cookieStore;
     }
 
+    public void resetStepCookies()
```
[the remark this answers is not in the document]
Comment (abudevich) [reply vividus-plugin-rest-api/src/main/java/org/vividus/http/CookieStoreProvider.java:42]: added `CookieStepMonitor`, which calls this method after execution step

File: vividus-plugin-rest-api/src/test/java/org/vividus/http/CookieStepMonitorTests.java [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+class CookieStepMonitorTests
+{
+    @Mock private CookieStoreProvider cookieStoreProvider;
+    @InjectMocks private CookieStepMonitor cookieStepMonitor;
+
+    @Test
+    void shouldClearCookiesAfterStep()
+    {
+        cookieStepMonitor.afterPerforming(null, false, null);
+        verify(cookieStoreProvider, times(1)).resetStepCookies();
```
Comment (valfirst) [inline vividus-plugin-rest-api/src/test/java/org/vividus/http/CookieStepMonitorTests.java:38]:
```suggestion
        verify(cookieStoreProvider).resetStepCookies();
```

File: vividus-plugin-rest-api/src/main/java/org/vividus/http/CookieStepMonitor.java [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+
+    public CookieStepMonitor(CookieStoreProvider cookieStoreProvider)
+    {
+        this.cookieStoreProvider = cookieStoreProvider;
+    }
+
+    @Override
+    public void afterPerforming(String step, boolean dryRun, Method method)
+    {
+        cookieStoreProvider.resetStepCookies();
```
Comment (valfirst) [inline vividus-plugin-rest-api/src/main/java/org/vividus/http/CookieStepMonitor.java:35]: I guess it should be reset only if it is not dry run

**Review by valfirst — CHANGES_REQUESTED** [review]

File: vividus-plugin-rest-api/src/main/java/org/vividus/http/CookieStepMonitor.java (hunk 5 above)
Comment (abudevich) [reply vividus-plugin-rest-api/src/main/java/org/vividus/http/CookieStepMonitor.java]: thanks. updated

**Pull request merged**